### DISCUSSION TITLE
Multiple selection rotation

### DISCUSSION THREAD

**Discussion by eko234** [ki-editor/ki-editor#317] [Ideas]
Hello, I come from kakoune, which I love with my heart, a thing that just wont let me use any other ide or editor is the idea of selections, using s form select matching and S for select spliting, and the capability to not transpose but to rotate the values of the selections. Usually I only rotate between 2 elements, which afik would be equivalent to the x keybinding in ki for swap, but I think rotation is a more powerful concept, just wanted to know what you think about it or if it is possible with the current primitves.

**Comment by wongjiahau** [comment]
Hey @eko234, do you have any examples of using rotation? I think rotation can be fairly easy to implement, but it needs to have specific use cases that cannot be achieved easily with other primitives.
